File: codegen/emitters/import_.py
from codegen.util import CCONV
from parser.ast import FunctionImportNode, ir

def emit(self, node: FunctionImportNode):
	if type(node) == FunctionImportNode:
		f = self._module.globals.get(node.name)
		if not f:
			f = ir.FunctionType(node.returnType[0], (T for T, _ in node.args.values()), node.vararg)
		return f
	else:
		return self.panic("'%s' is not a valid import", type(node).__name__[:-4])

__all__ = ["emit"]

File: lang/codegen/emitters/import.cpp
#include "codegen/codegen.h"

#include <fstream>
#include <filesystem>

#include "core/tea.h"
#include "frontend/lexer/Lexer.h"
#include "frontend/parser/Parser.h"

namespace fs = std::filesystem;

namespace tea {

	mir::Function* CodeGen::emitFunctionImport(const AST::FunctionImportNode* node) {
		tea::vector<Type*> params;
		for (const auto& [ty, _] : node->params)
			params.emplace(ty);

		mir::Function* func = module->addFunction(node->name, ctx.types.Function(node->returnType, params, node->vararg));
		func->storage = AST::StorageClass::Public;
		func->subclassData = node->extra;
		return func;
	}

} // namespace tea
